ps_checkout.php

```diff
@@ -1044,4 +1054,16 @@ public function hookDisplayAdminOrderMainBottom(array $params)
 
         return $this->display(__FILE__, '/views/templates/hook/displayAdminOrderMainBottom.tpl');
     }
+
+    public function hookReceiveWebhook_ps_checkout(array $params)
```

Comment:
Nom de la fonction très important et totalement régit par "hookReceiveWebhook_${module->name}"

Reply:
du coup idéalement dans le [README de ps_accounts](https://github.com/PrestaShopCorp/ps_accounts/edit/master/Readme.md)

Reply:
There are other way to specify what module should be targeted by a specific module. What do you think about restricting the hook names we could register (i.e `hookActionReceiveWebhook`) ?

### Solution 1: Restrict the module on the caller side

When calling Hook::exec, there is a parameter which allows you to filter what module will be called.

*classes/Hook.php:*
```php
    public static function exec(
        $hook_name,
        $hook_args = [],
        $id_module = null, // <-- The param I'm talking about
        $array_return = false,
        $check_exceptions = true,
        $use_push = false,
        $id_shop = null,
        $chain = false
```

## Solution 2:

Allow the hook to be dispatched everywhere, but send the primary target as a parameter.

When triggering the hook, we could have the main module in the args:

```php
$hookArgs = [
    'service' => 'ps_checkout',
    // ...
];

Hook::exec('actionReceiveWebhook', $hookArgs);
```

Then, on the ps_checkout side:

```php
public function hookActionReceiveWebhook(array $params)
{
    if ($params['service'] !== $this->name) {
        return;
    }
}
```